### PR TITLE
feat: add tags to conversations on removal

### PR DESCRIPTION
**Pull request by miketheman**
Create `add_tag` to set a tag to a Help Scout conversation.

We can use the added tag to trigger automated workflows.

### CODE REVIEW

**Review by ewdurbin — APPROVED** [review]

File: warehouse/helpdesk/services.py [hunk 1]
```diff
@@ -120,3 +124,32 @@ def create_conversation(self, *, request_json: dict) -> str:
...
+        tags.append(tag)
+
+        resp = self.http.put(
+            f"{conversation_url}/tags",
+            headers={"Authorization": f"Bearer {self.bearer_token}"},
+            json={"tags": tags},
+            timeout=10,
+        )
+        resp.raise_for_status()
+        return
```
Comment (di) [inline warehouse/helpdesk/services.py:155]: Driveby comment, but based on my read of https://api.helpdesk.com/docs#operation/ticketUpdateAddTag it looks like this could just be:

```suggestion
        resp = self.http.put(
            f"{conversation_url}/tags",
            headers={"Authorization": f"Bearer {self.bearer_token}"},
            json={"tags": [tag]},
            timeout=10,
        )
        resp.raise_for_status()
        return
````

I suspect the API doesn't drop tags that already exist but aren't included in the request.

Comment (miketheman) [reply warehouse/helpdesk/services.py:155]: @di Thanks, but that's the API docs for Help Desk, not Help Scout.
